src/libraries/System.Text.Json/gen/Reflection/TypeExtensions.cs

```diff
@@ -57,6 +55,22 @@ public static bool IsNullableValueType(this Type type, out Type? underlyingType)
             return false;
         }
 
+        public static bool IsNullableValueType(this Type type, out Type? underlyingType)
+        {
+            if (type.IsGenericType && type.Name.StartsWith("Nullable`1"))
+            {
+                underlyingType = type.GetGenericArguments()[0];
+                return true;
+            }
+
+            underlyingType = null;
+            return false;
+        }
+
+        public static bool IsObjectType(this Type type) => type.FullName == "System.Object";
+
+        public static bool IsStringType(this Type type) => type.FullName == "System.String";
```

Comment:
Is this a "good enough" check? Should we be using `Compilation.GetSpecialType(SpecialType.String_String)` instead?

See [https://github.com/dotnet/runtime/pull/51064#discussion_r615516155](https://github.com/dotnet/runtime/pull/51064#discussion_r615516155)

Reply:
Nope it's not good enough but needed to unblock for this PR. `Compilation.GetSpecialType(SpecialType.System_String)` looks like the way to go. I'm prepping a follow for this and other feedback.